File: 프로그래머스/unrated/181861. 배열의 원소만큼 추가하기/배열의 원소만큼 추가하기.py
def solution(arr):
    X = [] #빈배열 X를 생성합니다.
    for a in arr: #배열 arr의 각원소에 대해서
        X.extend([a] * a) #원소a를 a번 배열X의 맨뒤에 추가합니다.
    return X #배열 X를 반환합니다.

File: 프로그래머스/0/181861. 배열의 원소만큼 추가하기/배열의 원소만큼 추가하기.java
import java.util.*;

class Solution {
    public ArrayList<Integer> solution(int[] arr) {
        ArrayList<Integer> answer = new ArrayList<>();
        for(int a: arr){
            for(int i=0;i<a;i++){
                 answer.add(a);     
            }
        }
        return answer;
    }
}